### PR TITLE
refactor: NodeJS version not maintained deprecation

### PR DESCRIPTION
**Pull request by vaidyamilind**
Closes: #9586

Updated nodejs version minimum requirement to v12.

### CODE REVIEW

File: bin/serverless.js [hunk 1]
```diff
@@ -7,7 +7,7 @@ EvalError.$serverlessCommandStartTime = process.hrtime();
 
 const nodeVersion = Number(process.version.split('.')[0].slice(1));
 
-if (nodeVersion < 10) {
+if (nodeVersion < 12) {
```
Comment (medikoo) [inline bin/serverless.js:10]: @vaidyamilind  we should not do that, as that removes support for v10, while in context of this we should just deprecate it.

Please see link in an issue description that suggests how to configure the deprecation

File: docs/deprecations.md [hunk 2]
```diff
@@ -384,9 +384,9 @@ Configure `httpApi.payload` explicitly to ensure seamless migration.
 
 Deprecation code: `OUTDATED_NODEJS`
 
-Support for Node.js v6 and v8 will be dropped with v2.0.0 release
+Support for Node.js v10 will be dropped with v3.0.0 release
```
Comment (medikoo) [inline docs/deprecations.md:387]: This message should remain valid for v1 users, which may also come to see documentation for this deprecation.

I'd probably reword it to:

```
Support for Node.js v8 was dropped with v2 release, while support for Node.js v10 will be dropped with v3 release
```

**Review by medikoo — CHANGES_REQUESTED** [review]
Thank you @vaidyamilind ! Please see my commments

File: bin/serverless.js (hunk 1 above)
Comment (vaidyamilind) [reply bin/serverless.js:10]: Currently, for all < v10, we are showing Initialization error and exiting.
Do we need an OUTDATED_NODEJS log for version within range v10 and v12? or for all version which is < 12

Comment (medikoo) [reply bin/serverless.js:10]: Please see an issue description. I've linked there specifically an implementation we expect.

In v2 we just need to show deprecation log, and let v10 deployments work without issues

Comment (vaidyamilind) [reply bin/serverless.js:10]: @medikoo Updated

**Review by medikoo — APPROVED** [review]
Looks great @vaidyamilind ! Thank you
